zorkington-game.js

```diff
@@ -0,0 +1,503 @@
+class House {
+    constructor() {
+        this.roomCollection = [];
+    }
+
+    addRoom(room) {
+        this.roomCollection.push(room);
+    }
+
+    // helper function to get room based on column and row
+    getRoom(col, row) {
+        let currRoom = null;
+
+        this.roomCollection.forEach((room) => {
+            if (col === room.column &&
+                row === room.row) {
+                currRoom = room;
+            }
+        })
+        return currRoom;
+    }
+
+    getEntryRoom() {
+        return this.roomCollection[0];
+    }
+}
+
+class Room {
+    constructor(roomNumber, column, row, name, description, takeableItemDesc, locked) {
+        this.roomNumber = roomNumber;
+        this.column = column;
+        this.row = row;
+        this.name = name;
+        this.description = description;
+        this.takeableItemDescription = takeableItemDesc;
+        this.fullDescription = description + takeableItemDesc;
+        this.locked = locked;
+    }
+
+    unlock() {
+        this.locked = false;
+    }
+
+    isOpen() {
+        return !this.locked;
+    }
+
+}
+
+class ItemCollection {
+    constructor() {
+        this.items = [];
+    }
+
+    // Add an item into array to store defined items
+    addItem(item) {
+        this.items.push(item);
+    }
+
+    // find an item in a room and return it
+    getItem(itemWanted) {
+        const index = this.items.findIndex(item => item.name === itemWanted);
+        return this.items[index];
+    }
+}
+
+class Item {
+    constructor(roomNumber, name, desc, takeable, hasFiniteLife, affectsRoomLock, affectsPlayerHealth, life) {
+        this.roomNumber = roomNumber;
+        this.name = name;
+        this.description = desc;
+        this.takeable = takeable;
+        this.hasFiniteLife = hasFiniteLife;
+        this.useStatus = false;
+        this.affectsRoomLock = affectsRoomLock;
+        this.affectsPlayerHealth = affectsPlayerHealth;
+        this.life = life;
+    }
+}
+
+class Player {
+
+    constructor(name) {
+        this.name = name;
+        this.inventory = [];
+        this.healthLevel = 100;
+        this.isBandaged = false;
+        this.currentRoom = house.getEntryRoom();
+        this.currentColumn = 1;
+        this.currentRow = 1;
+    }
+
```

Comment:
If you wanted to refactor this, since there's only going to be one player object, you could have just coded this as an object rather then a class/constructor.

Constructors are great when trying to pump out a lot of the same things with the same attributes but if there's only one of them you can just create a `let player ={...}`